stub_uploader/metadata.py

```diff
@@ -141,13 +141,32 @@ def strip_types_prefix(dependency: str) -> str:
 def verify_typeshed_req(req: Requirement) -> None:
     if not req.name.startswith(TYPES_PREFIX):
         raise InvalidRequires(f"Expected dependency {req} to start with {TYPES_PREFIX}")
+
     if not canonical_name(req.name) in uploaded_packages.read():
         raise InvalidRequires(
             f"Expected dependency {req} to be uploaded from stub_uploader"
         )
 
+    # TODO: make sure that if a typeshed distribution depends on other typeshed stubs,
+    # the upstream depends on the upstreams corresponding to those stubs.
+    # See https://github.com/typeshed-internal/stub_uploader/pull/61#discussion_r979327370
+
+
+# Presence in the top 1000 PyPI packages could be a necessary but not sufficient criterion for
+# inclusion in this allowlist.
+# Note we could loosen our criteria once we address:
+# https://github.com/typeshed-internal/stub_uploader/pull/61#discussion_r979327370
+EXTERNAL_REQ_ALLOWLIST = {
```

Comment:
`pytorch` would also be useful for at least one package, off the top of my head

```suggestion
EXTERNAL_REQ_ALLOWLIST = {
    "pytorch",
```

Reply:
Sure, although note that the dist name is torch ;-)

Reply:
D3DShot uses torch.

PyInstaller could be simplified by having access to modulegraph.